src/roswire/common/source.py

```diff
@@ -303,13 +333,15 @@ def __process_python_executables(
             if program.startswith("nodes/"):
                 name = Path(program[0]).stem
                 sources: t.Set[str] = set()
-                if 'cwd' in cmake_env:
-                    sources = set()
-                    sources.add(os.path.join(cmake_env['cwd'], program))
-                else:
-                    sources.add(program)
+                source = self._resolve_to_real_file(program, package, cmake_env)
+                sources.add(source)
                 logger.debug(f"Adding Python sources for {name}")
                 executables[name] = CMakeTarget(name,
                                                 SourceLanguage.PYTHON,
                                                 sources,
                                                 set())
+
+    def _list_to_dict(self, values: t.List[str]) -> t.Dict[str, str]:
```

Comment:
How does a list get converted to a dict? Seems like there's more going on here? It looks like the list is of the form: `[k1, v1, k2, v2, ...]`? Maybe rename this to `_key_val_list_to_dict`?